elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchProject.java

```diff
@@ -68,6 +68,10 @@
       final String name = pair.right;
       final String expr = pair.left.accept(translator);
 
+      if (ElasticsearchRules.isItem(pair.left)) {
+        implementor.addExpressionItemMapping(name, ElasticsearchRules.stripQuotes(expr));
+      }
+
       if (expr.equals("\"" + name + "\"")) {
         fields.add(name);
```

Comment:
After identifying the fields necessary you could store them in a list in the implementor so that other operators (e.g., Sort) can take advantage of them.